src/main/kotlin/org/rust/cargo/runconfig/command/RunClippyAction.kt

```diff
@@ -5,17 +5,38 @@
 
 package org.rust.cargo.runconfig.command
 
+import com.intellij.notification.NotificationType
 import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.Computable
 import org.rust.cargo.icons.CargoIcons
+import org.rust.cargo.project.settings.toolchain
 import org.rust.cargo.runconfig.getAppropriateCargoProject
 import org.rust.cargo.toolchain.CargoCommandLine
 import org.rust.cargo.toolchain.run
+import org.rust.ide.actions.InstallComponentAction
+import org.rust.ide.notifications.showBalloon
 
 class RunClippyAction : RunCargoCommandActionBase(CargoIcons.CLIPPY) {
-
     override fun actionPerformed(e: AnActionEvent) {
         val project = e.project ?: return
         val cargoProject = getAppropriateCargoProject(e.dataContext) ?: return
+        val projectDirectory = cargoProject.manifest.parent ?: return
+
+        val isClippyInstalled = ProgressManager.getInstance().runProcess(Computable {
+            val rustup = project.toolchain?.rustup(projectDirectory) ?: return@Computable true
```

Comment:
Let's add some comment that we don't handle (and probably don't want) cases when a user doesn't have `rustup` or has too old `rustup` (without `clippy` component)